src/Orleans.Core.Abstractions/Activation/IGrainContext.cs

```diff
@@ -50,21 +50,39 @@ public interface IGrainContext : ITargetHolder, IEquatable<IGrainContext>
         /// <param name="value">The component instance.</param>
         void SetComponent<TComponent>(TComponent value);
 
+        /// <summary>
+        /// Submits an incoming message to this instance.
+        /// </summary>
+        /// <param name="message">The message.</param>
         void ReceiveMessage(object message);
 
         IWorkItemScheduler Scheduler { get; }
-        PlacementStrategy PlacementStrategy { get; }
 
+        /// <summary>
+        /// Start activating this instance.
+        /// </summary>
+        /// <param name="requestContext">The request conext of the request which is causing this instance to be activated, if any.</param>
+        /// <param name="cancellationToken">A cancellation token which when cancelled, indicates that the process should complete prompty.</param>
         void Activate(Dictionary<string, object> requestContext, CancellationToken? cancellationToken = default);
-        void Deactivate(CancellationToken? cancellationToken = default);
+
+        /// <summary>
+        /// Start deactivating this instance.
+        /// </summary>
+        /// <param name="deactivationReason">The reason for deactivation, for informational purposes.</param>
+        /// <param name="cancellationToken">A cancellation token which when cancelled, indicates that the process should complete prompty.</param>
+        void Deactivate(DeactivationReason deactivationReason, CancellationToken? cancellationToken = default);
```

Comment:
Obviously, it should, but this PR currently also plumbs through cancellation tokens for activate/deactivate as well as the deactivation reason

Reply:
This ought to be split into a separate PR